modules/custom/express_dashboards/express_dashboard_lock/express_dashboard_lock.module

```diff
@@ -0,0 +1,241 @@
+  );
+  $form['lock_leave_message'] = array(
+    '#type' => 'textarea',
+    '#title' => t('Leave a message for locked out users'),
+    '#default_value' => variable_get('lock_leave_message', t('This site has been locked. For more information contact the Site Owner.')),
+  );
+  $form['submit'] = array(
+    '#type' => 'submit',
+    '#value' => t('Save configuration'),
+  );
+  return $form;
+}
+
+/**
+ * Implements hook_form_submit().
+ *
+ * You can't lock yourself out while submitting the form.
+ */
+function express_dashboard_lock_admin_form_submit($form, &$form_state) {
+  global $user;
+  // Exclude unnecessary elements.
+  form_state_values_clean($form_state);
+
+  foreach ($form_state['values'] as $key => $value) {
+    if (is_array($value) && isset($form_state['values']['array_filter'])) {
+      $value = array_keys(array_filter($value));
+    }
+    variable_set($key, $value);
+  }
+
+  $dont_lock_yourself_out_when_submitting = $form_state['values']['lock_allow_users'];
+  $dont_lock_yourself_out_when_submitting .= t(' ') . $user->name;
+
+  $allowed_users = explode(' ', variable_get('lock_allow_users'));
+
+  if (!in_array($user->name, $allowed_users)) {
+    variable_set('lock_allow_users', $dont_lock_yourself_out_when_submitting);
+  }
+
+  drupal_set_message(t('Lock has been enabled.'));
+}
+
+/**
+ * Implements hook_permission().
+ */
+function express_dashboard_lock_permission() {
+  return array(
+    'dashboard user lock' => array(
+      'title' => 'Lock users via the dashboard',
+      'description' => 'Allows you to lock out users.',
+    ),
+  );
+}
+
+/**
+ * Implements hook_secure_permissions().
+ */
+function express_dashboard_lock_secure_permissions($role) {
+  $permissions = array(
+    'administrator' => array(
+      'dashboard user lock',
+    ),
+    'developer' => array(
+      'dashboard user lock',
+    ),
+    'site_owner' => array(
+      'dashboard user lock',
+    ),
+  );
+
+  if (isset($permissions[$role])) {
+    return $permissions[$role];
+  }
+}
+
+/**
+ * Check to see if user->name is in lock_allow_users variable.
+ */
+function _lock_allow_users_check($username) {
+  $output = FALSE;
+  $allowed_users = explode(' ', variable_get('lock_allow_users'));
+
+  if (in_array($username, $allowed_users)) {
+    $output = TRUE;
+  }
+  return $output;
+}
+
+/**
+ * Check roles to see if they have access.
+ */
+function _lock_by_role_check() {
+  global $user;
+  $output = FALSE;
+  $user_roles = variable_get('lock_by_role');
+
+  foreach (array_keys($user->roles) as $rid) {
+    if (in_array($rid, $user_roles)) {
+      $output = TRUE;
+    }
+  }
+  return $output;
+}
+
+/**
+ * Locked or not? If this returns false, you may pass.
+ */
+function _lock_final_check() {
+  global $user;
+  $output = FALSE;
+
+  if (isset($user->name)) {
+    if (!user_has_role(5) && !_lock_allow_users_check($user->name) && _lock_by_role_check()) {
```

Comment:
We can't assume the role ids will always be the same. Use user_role_load_by_name() to get the role id to check.